### ISSUE TITLE
VB Casing behavior difference between VS 2013 and Roslyn

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#2909]
[A customer reports in DevDiv 1172900]

I have a solution that I can build in VS 2013 without any problem. However, when I try and build the same solution in VS 2015 it will not compile.
I'm not 100% sure if this is a bug with the compiler or not. The project that is having trouble building in VS 2015 is a VB project. The code that is having an error is: 
```vb
Case MySqlDbType.Datetime 
```
The compile error is BC31429. The problem is that the MySql.Data dll that I am referencing contains an enum with the following two members:
```vb
MySqlDbType.DateTime 
MySqlDbType.Datetime 
```
VS2013 seems to handle the case differences just fine even though VB is a case insensitive language. The VS2015 compiler, however, does not seem to handle the case differences the same way and gives me an ambiguity error.

**Comment by gafter** [comment]
@AnthonyDGreen @ljw1004 @AlekseyTs Do you have any comment on the behavior change? What should the compiler do in this case?

**Comment by ljw1004** [comment]
We spent lots of LDM discussing case issues for identifier lookup. We even went so far as to discuss what case should be used when you overload a method. (answer: it should use the case of the thing it's overloading rather than the case you write). And what case should be used when you overload a method that has two differently-cased existing overloads (answer: I forget).

I'm surprised, given how careful we were about those edge cases, that we didn't think about this big case. I think the compiler should resolve "ambiguity-in-case-only" by picking the identical case, if there is one, as its final step in ambiguity resolution.

**Comment by gafter** [comment]
After a bit of experimenting, it appears that the old compiler selected arbitrarily which enum member to bind to. Adding new enum members of other names or changing the order of the declarations would change which one would be bound to.

That seems particularly dangerous. If we want to implement this as a compatibility concession, we should at least produce a warning. I do not believe it would be practical to try to reproduce the same hash function used in the native compiler to cause it to bind the same way it used to.

**Comment by AlekseyTs** [comment]
I believe behavior of VS 2013 compiler is not intentional and is not deterministic. Roslyn does the right thing, we should consider keeping this behavior.

**Comment by gafter** [comment]
This is being discussed by the Compat Council.

Or perhaps it is the compat council who is discussing it...

**Comment by gafter** [comment]
In the user's test case, the two enum constants have the same value, and the second one is marked [Obsolete]. I'll investigate accepting such programs and always picking the first member found of the given name (only in the particular case of conflicting enum constants that have the same value).

**Comment by gafter** [comment]
@AlekseyTs Please verify.

**Comment by gafter** [comment]
After further review, it appears that the old compiler is indeed deterministic. It always picks the (lexically) first one. And there are public APIs that depend on that. See, for example, 

http://lists.mysql.com/commits/48012

**Comment by gafter** [comment]
Now that we know there are many users (of the MySql .net APIs) who would be affected by this, sending it back to shiproom for reconsideration for version 1.0.

The fix #3056 was previously reviewed, tested, and verified before it was reverted in #3373.

**Comment by gafter** [comment]
Fixed via f23342c44eea9384d8bf619772329aa9244eefb5 .

@AlekseyTs Please Verify.

**Comment by AlekseyTs** [comment]
Verified.
